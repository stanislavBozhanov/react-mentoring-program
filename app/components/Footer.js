import React from 'react';
import PropTypes from 'prop-types';


const Footer = (props) => <footer>
  <span>{props.appName}</span>
</footer>;

Footer.propTypes = {
  appName: PropTypes.String
};

export default Footer;
