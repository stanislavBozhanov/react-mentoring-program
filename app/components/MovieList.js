import React, { Component } from 'react';
import Movie from "./Movie";


export default class MovieList extends Component {
  render() {
    return (
      <div>
        <ul>
          {this.props.movies.map((movie) => <Movie key={movie.name} movie={movie} />)}
        </ul>
      </div>
    )
  }
}
