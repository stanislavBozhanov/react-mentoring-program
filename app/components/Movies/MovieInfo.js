import React, { Component } from 'react';


export default class MovieInfo extends Component {
  render() {
    return (
      <div>
        <div>Title</div>
        <div>Sub Title</div>
        <div>Cover photo</div>
        <div>Year and duration</div>
        <div>Description</div>
      </div>
    )
  }
}
