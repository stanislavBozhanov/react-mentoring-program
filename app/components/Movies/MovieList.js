import React, { Component } from 'react';
import { connect } from 'react-redux';
import Movie from "./Movie";


class MovieList extends Component {
  render() {
    return (
      <div>
        <ul>
          { this.props.movies ?
            this.props.movies.map((movie) => <Movie key={movie.id} movie={movie} />) :
            'No similar movies found.'
          }
        </ul>
      </div>
    )
  }
}

function mapStateToProps(state) {
  return {
    movies: state.moviesFiltered
  }
}

export default connect(mapStateToProps)(MovieList);
