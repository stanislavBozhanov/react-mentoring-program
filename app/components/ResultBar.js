import React, { Component } from 'react';


export default class ResultBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sort: null,
      genre: null
    }
  }

  render() {
    return (
      <div></div>
    );
  }
}
