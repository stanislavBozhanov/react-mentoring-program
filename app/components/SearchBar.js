import React, { Component } from 'react';


export default class SearchBar extends Component {
  searchText = React.createRef();

  searchMovies = () => {
    this.props.search(this.searchText.current.value)
  };

  render() {
    return (
      <div className="search-bar">
        <span>{this.props.appName}</span>
        <h2>Find your movie</h2>
        <input type="text" name="searchText" ref={this.searchText}/>
        <div>
          <span>Search by</span>
          <button>Title</button>
          <button>Genre</button>
          <button onClick={this.searchMovies}>Search</button>
        </div>
      </div>
    )
  }
}
