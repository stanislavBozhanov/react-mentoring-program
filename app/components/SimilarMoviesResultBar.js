import React, { Component } from 'react';


export default class SimilarMoviesResultBar extends Component {
  render() {
    return (
      <div>
        <span>{this.props.infoText}</span>
      </div>
    )
  }
}
