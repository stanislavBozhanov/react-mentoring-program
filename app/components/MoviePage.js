import React, { Component } from 'react';
import MovieInfo from "./MovieInfo";
import SimilarMoviesResultBar from "./SimilarMoviesResultBar";
import MovieList from "./MovieList";
import Footer from "./Footer";


export default class MoviePage extends Component {
  render() {
    return (
      <React.Fragment>
        <MovieInfo/>
        <SimilarMoviesResultBar/>
        <MovieList/>
        <Footer/>
      </React.Fragment>
    );
  }
}
