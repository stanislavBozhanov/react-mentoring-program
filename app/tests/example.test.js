import React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import Footer from "../components/Footer";
import ResultBar from "../components/ResultBar";
import SearchBar from "../components/SearchBar";
import SimilarMoviesResultBar from "../components/SimilarMoviesResultBar";
import SortableResultBar from "../components/SortableResultBar";
import MovieInfo from "../components/Movies/MovieInfo";
import MovieList from "../components/Movies/MovieList";
import MoviePage from "../components/Movies/MoviePage";


// Setup Enzyme's react adapter
Enzyme.configure({adapter: new Adapter()});


it('Renders Footer component', () => {

  const tree = TestRenderer.create(<Footer/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders ResultBar component', () => {

  const tree = TestRenderer.create(<ResultBar/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders SearchBar component', () => {

  const tree = TestRenderer.create(<SearchBar/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders SimilarMoviesResultBar component', () => {

  const tree = TestRenderer.create(<SimilarMoviesResultBar/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders SortableResultBar component', () => {

  const tree = TestRenderer.create(<SortableResultBar/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders MovieInfo component', () => {

  const tree = TestRenderer.create(<MovieInfo/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders MovieList component', () => {

  const tree = TestRenderer.create(<MovieList/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders MoviePage component', () => {

  const tree = TestRenderer.create(<MoviePage/>).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should change data', () => {
  const eventMock = {
    target: {
      value: 'Test value'
    }
  };

  const component = shallow(<SearchBar/>);
  component.find('input').simulate('change', eventMock);

  console.log(component);
});

