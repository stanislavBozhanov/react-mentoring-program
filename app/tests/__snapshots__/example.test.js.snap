// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Footer component 1`] = `
<footer>
  <span />
</footer>
`;

exports[`Renders MovieInfo component 1`] = `
<div>
  <div>
    Title
  </div>
  <div>
    Sub Title
  </div>
  <div>
    Cover photo
  </div>
  <div>
    Year and duration
  </div>
  <div>
    Description
  </div>
</div>
`;

exports[`Renders MovieList component 1`] = `
<div>
  <ul>
    No similar movies found.
  </ul>
</div>
`;

exports[`Renders MoviePage component 1`] = `
Array [
  <div>
    <div>
      Title
    </div>
    <div>
      Sub Title
    </div>
    <div>
      Cover photo
    </div>
    <div>
      Year and duration
    </div>
    <div>
      Description
    </div>
  </div>,
  <div>
    <span />
  </div>,
  <div>
    <ul>
      No similar movies found.
    </ul>
  </div>,
]
`;

exports[`Renders ResultBar component 1`] = `<div />`;

exports[`Renders SearchBar component 1`] = `
<div
  className="search-bar"
>
  <span />
  <h2>
    Find your movie
  </h2>
  <input
    name="searchText"
    type="text"
  />
  <div>
    <span>
      Search by
    </span>
    <button>
      Title
    </button>
    <button>
      Genre
    </button>
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`Renders SimilarMoviesResultBar component 1`] = `
<div>
  <span />
</div>
`;

exports[`Renders SortableResultBar component 1`] = `
<div>
  <span />
  <div>
    <span>
      Sort by
    </span>
    <button>
      release date
    </button>
    <button>
      rating
    </button>
  </div>
</div>
`;
